benchmarks/dataset_serialize_benchmark.py

```diff
@@ -0,0 +1,232 @@
+import itertools
+import logging
+import os
+import shutil
+import time
+import uuid
+
+import conbench.runner
+import pyarrow
+import pyarrow.dataset as ds
+
+from benchmarks import _benchmark
+
+log = logging.getLogger(__name__)
+
+
+@conbench.runner.register_benchmark
+class DatasetSerializeBenchmark(_benchmark.Benchmark):
+    """
+    This benchmark is supposed to measure the time it takes to write data from
+    memory (from a pyarrow Table) to a tmpfs file system, given a specific
+    serialization format (parquet, arrow, ...).
+
+    To make this benchmark agnostic to disk read performance on the input side
+    of things, the data is read fully into memory before starting (and timing)
+    the benchmark function. That is (believed to be) achieved with:
+
+        data = source_dataset.to_table(filter=somefilter, ...)
+
+    After data of interest has been read into memory, the following call is
+    used for both serialization and writing-to-filesystem in one go:
+
+        pyarrow.dataset.write_dataset(format=someformat, ...)
+
+    That operation is timed (and the duration is the major output of this
+    benchmark).
+
+    The data is written to `/dev/shm` (available on all Linux systems). This is
+    a file system backed by RAM (tmpfs). The assumption is that writing to
+    tmpfs is fast (so fast that benchmark duration is significantly affected by
+    the serialization itself), and stable (so that its performance is ~constant
+    across runs on the same machine).
+
+    This benchmark does not resolve how much time goes into the CPU work for
+    serialization vs. the system calls for writing to tmpfs (that would be a
+    different question to answer, an interesting one, that is maybe more of a
+    task for profiling).
+
+    There are two dimensions that are varied:
+
+        - serialization format
+        - amount of the data being written, as set by a filter on the input
```

Comment:
Ah I see, the filter is the means by which we are varying the data size

Reply:
Right. It's a little indirect, and we don't get to see how big the data actually was. I think this can/should be improved in the future.

Reply:
update: with the recent commits I changed the approach